cosmwasm/packages/wasmi-runtime/src/wasm/errors.rs

```diff
@@ -60,7 +62,8 @@ impl From<WasmEngineError> for EnclaveError {
             MemoryAllocationError => EnclaveError::MemoryAllocationError,
             MemoryReadError => EnclaveError::MemoryReadError,
             MemoryWriteError => EnclaveError::MemoryWriteError,
-            NotImplemented => EnclaveError::NotImplemented,
+            BadQueryChainRequest => EnclaveError::BadQueryChainRequest,
+            BadQueryChainResponse => EnclaveError::BadQueryChainResponse,
             // Unexpected WasmEngineError variant
```

Comment:
So, in the cosmwasm implementation they return an error into the contract when this happens. This is especially important for `BadQueryChainResponse` because if contract B misbehaves, we don't want to just crash contract A.
Let's talk about this and thing about the best way to implement this.

Reply:
It turns out we only use `BadQueryChainRequest`...

Reply:
Let me know what you think about 3f9b2da8a6ede4cfc32f8ea69a62dc7f316e49a0

Reply:
Ah yeah, I didn't comment - it's great :) :+1: